src/report/ReportPoll.ts

```diff
@@ -0,0 +1,67 @@
+import { MatrixClient } from "matrix-bot-sdk";
+import { ReportManager } from './ReportManager';
+
+class InvalidStateError extends Error {}
+
+export class ReportPoll {
+    private _client: MatrixClient;
+    private _manager: ReportManager;
+    private _save: (a: number) => Promise<any>;
+    private _from = 0;
+
+    private _interval: ReturnType<typeof setInterval> | null = null;
+
+    constructor(
+        client: MatrixClient,
+        manager: ReportManager,
+        save: (a: number) => Promise<any>
+    ) {
+        this._client = client;
+        this._manager = manager;
+        this._save = save;
+    }
```

Comment:
I think you can do `private readonly client: MatrixClient` in the constructor arguments and it'll be the same as an instance variable if you didn't know that, though I am not sure whether it is a good idea